projects/observability/src/shared/dashboard/data/graphql/specifiers/explore-selection-specification.model.ts

```diff
@@ -19,10 +20,10 @@ export class ExploreSelectionSpecificationModel
   @ModelProperty({
     key: 'metric',
     displayName: 'Metric',
-    type: STRING_PROPERTY.type,
+    type: UNKNOWN_PROPERTY.type,
     required: true
   })
-  public metric!: string;
+  public metric!: string | AttributeExpression;
```

Comment:
Can you give some contexts around these changes?

Reply:
<img width="665" alt="Screenshot 2022-08-03 at 12 06 09 AM" src="https://user-images.githubusercontent.com/101543426/182448681-514d3bad-df7a-4dc0-b3e6-f0b621f221ea.png">

For graphql we can pass `keyExpression` for various fields which is an object that has a key and a subpath, however we control the graphql using models and data sources. Some of the data sources/models that are updated in the PR only supported `keyExpression: {key: "key_value"}` but we needed support for `keyExpression: {key: "key_value", subpath: "subpath_value"}`
An example for this would be adding filters based on merchant ids. Here our key is tags and subpath is merchant. For this, I have extended the type of some fields to be an old type or AttributeExpression. These cases were handled in most of the models but I think since the models I have updated were not used enough or the team did not have such a requirement before it was missing. 

Why `UNKOWN_TYPE`  
Since that field can hold either a string value or an object we are setting it as a UNKNOWN_TYPE. These type definitions are used by hyperdash for validating whether the JSON passed to it is valid and adheres to the schema mentioned here.